src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,155 @@
+package duke;
+
+import java.time.LocalDate;
+
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.TaskList;
+import duke.task.TodoTask;
+import duke.ui.Ui;
+
+
+/**
+ * Represents a parser that parses user input.
+ */
+public class Parser {
+
+    /** Represents the list of tasks. */
+    private static TaskList tasks = new TaskList();
+
+    /** Represents the type of task. */
+    private enum TaskType {
+        TODO, DEADLINE, EVENT
+    }
+
+    /** Represents the type of modification to be made. */
+    private enum ModifyTask {
+        MARK, UNMARK, DELETE
+    }
+
+    /**
+     * Adds a Task object to the specified list of tasks.
+     * @param tsk The type of task to be added.
+     * @param input The user input.
+     * @throws DukeException If the user input is invalid.
+     */
+    private static String addTask(TaskType tsk, String input) throws DukeException {
```

Comment:
should `tsk` be renamed to "task" for better readability?

Reply:
I agree that it should be named task. Thanks :)